File: src/my_cpp_pkg/src/number_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int64.hpp"

class NumberPublisherNode : public rclcpp::Node
{
public:
    NumberPublisherNode() : Node("number_publisher")
    {
        // Parameter name is the first argument, and the default value
        // is the second argument.  The type for the parameter is
        // inferred from the default value.
        this->declare_parameter("number_to_publish", 2);
        this->declare_parameter("frequency", 1.0);

        number_ = this->get_parameter("number_to_publish").as_int();
        double frequency = this->get_parameter("frequency").as_double();

        publisher_ = this->create_publisher<example_interfaces::msg::Int64>("number", 10);
        timer_ = this->create_wall_timer(std::chrono::milliseconds((int)(1000.0 / frequency)),
                                         std::bind(&NumberPublisherNode::publishNumber, this));
        RCLCPP_INFO(this->get_logger(), "Number publisher has been started.");
    }

private:
    void publishNumber()
    {
        auto msg = example_interfaces::msg::Int64();
        msg.data = number_;
        publisher_->publish(msg);
    }

    rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;

    int number_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<NumberPublisherNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/number_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from example_interfaces.msg import Int64


class NumberPublisherNode(Node):
    def __init__(self):
        super().__init__('number_publisher')
        self.declare_parameter('number_to_publish', 2)
        self.declare_parameter('frequency', 1.0)
        
        self.number_ = self.get_parameter('number_to_publish').value
        self.frequency_ = self.get_parameter('frequency').value

        self.get_logger().info(f'number being published is {self.number_}')

        self.publisher_ = self.create_publisher(Int64, 'number', 10)
        self.timer_ = self.create_timer(self.frequency_, self.publish_number)
        self.get_logger().info('Number Publisher has been started')

    def publish_number(self):
        msg = Int64()
        msg.data = self.number_
        self.publisher_.publish(msg)


def main(args=None):
    rclpy.init(args=args)
    node = NumberPublisherNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == '__main__':
    main()
